### ISSUE TITLE
Mobile SMS friendly international phone format

### ISSUE THREAD

**Issue by rderharo** [Bluefieldscom/intl-tel-input#162]
Hi there, I use this plugin to allow users to enter SMS phone numbers. The problem I'm finding is that the plugin is adding a trunk prefix to Philippines phone numbers (and many other countries as well), and the issue is that it seems that mobile phone numbers should be in international format, not local format.

See this wikipedia article (https://en.wikipedia.org/wiki/Trunk_prefix):
> However, because the worlds telephone systems are "intelligent" enough to "know" from where the call has been originated, to where the call is directed and as to how it is to be charged, *if the full International number is used, then this full International number may be dialed from any telephone anywhere*. This is particularly important for users of Mobile (Cell) 'phones. **Such users are strongly advised to store all numbers in their 'phones in the form of "+ (Country Code) (Area Code) (Local Number)", no matter where they live**. Storing numbers in this manner will ensure that they can connect to any such stored number from any Area or Country in the world without any further consideration when they travel to any destination in the world.

Is it possible to have a flag that determines whether the intended phone number is going to be a mobile format, and as such enter everything as international format with no additional prefix added?

Edit: I might be oversimplifying the issue, but I'm trying to understand how I can allow users to enter their SMS numbers regardless of their location in the world.

**Comment by jackocnr** [comment]
I'm sorry I don't really understand. What version of the plugin are you using? What initialisation options are you using? By default, `nationalMode` is set to true, which means users can enter their national number, and then if you use the plugin's `getNumber` method, you get the full international number, which can be dialled from anywhere in the world (and will work for sending SMSs from anywhere in the world).

**Comment by rderharo** [comment]
I will try to explain with a specific example so you can try to reproduce it on your end.

I am using v5.0.1, by the way.

This is how the plugin is initialized:
```
telInput.intlTelInput({
      utilsScript: "/vendor/intl-tel-input/js/utils.js",
      preferredCountries: [],
      nationalMode: true
});
```

Mostly everything works as expected (thank you), but what I found is that for some locales (Philippines is an example) the actual phone number will be +63 917 555 1111 (as a fake example).

The user will select the country from the drop-down and enter their number like: 9175551111 and hit send. The plugin is taking that number, and outputs: 09175551111 (notice how it appended a `0` prefix?).

The 0 is the trunk prefix, and for mobile numbers it will not work when I try to use it to talk to an SMS aggregator that will send them a text message. From my understanding, the trunk prefix is only necessary for making local calls to land lines. What I'm trying to do with the number is send international mobile text messages, and from what I've read there should be no trunk prefix used; only country code, area code, and phone number.

**Comment by rderharo** [comment]
Okay so to update the situation, I realized that on the `getNumber()` method I was passing `NATIONAL` for the `numberFormat`. I have changed that to `INTERNATIONAL` and it correctly drops the trunk prefix.

However, the reason why I was passing `NATIONAL` was because I did not want the country code (I am retrieving that separately through `telInput.intlTelInput("getSelectedCountryData").dialCode;`. So I had to use regular expressions to remove the country code from the phone number string, as so:

```
regex = new RegExp('^' + mobileCountryCode);
phoneNum = phoneNum.replace(regex, '');
```

So far so good. Now the only problem I have is a *cosmetic problem*. When I call this: `telInput.intlTelInput('setNumber', '+' + number);` it adds the trunk prefix back in (`nationalMode` is on).

I guess there is no way I can get international phone format without disabling national mode? Because I want to be able to select the country from the dropdown without seeing the country code in the phone field, but I want to deal with international number formats. Does that make sense?

**Comment by jackocnr** [comment]
"I want to be able to select the country from the dropdown without seeing the country code in the phone field, but I want to deal with international number formats"

I feel like this is the default behaviour of the plugin. If you call setNumber with a full international number, it will use the country dial code to set the flag, and then convert the number into a national number (removing the country dial code) and display that. Then if you use getNumber, it will give you the full international number again.

**Comment by rderharo** [comment]
I agree that when it comes to country codes, the plugin behaves wonderfully. The issue appears when dealing with the trunk prefix, as the difference between the `national` and `international` formats (for me) means the difference of 09175551111 versus 9175551111 despite my calling `setNumber()` with "+639175551111".

Trunk prefixes are for local phone numbers only, while international phone numbers do not deal with trunk prefixes, correct? At least that's my understanding. So given that, when I call `getNumber()` with `INTERNATIONAL` for the `numberFormat` it returns me the number without the trunk prefix, and when I call `getNumber()` with `NATIONAL` for the `numberFormat`, it returns me the number *with* the trunk prefix.

But because I cannot define a number format when calling `setNumber` whatever number I pass, it always ends up in NATIONAL format when nationalMode is true, when I actually want to get and set in INTERNATIONAL format while nationalMode is true. Hope that makes sense...

**Comment by jackocnr** [comment]
The point of nationalMode is that users are most used to thinking of their phone numbers in a national context (with the trunk prefix) - that's what they have to dial everyday - that's how they remember their numbers. So IMO user's should only ever see (and enter) their numbers in that format (again, that's **with** the trunk prefix). Don't you think that's a better experience for the user?

Or am I misunderstanding the problem?

At the end you talk about setNumber, but I don't really understand. Can you give an example of what you're trying to do and what's currently happening? It sounds like you want to display an international number to the user, which I think would be confusing for them.

**Comment by rderharo** [comment]
To your first point, I can agree with that. If a user is familiar with entering the trunk prefix, then they should be able to enter it. The great thing about this plugin is that whether they enter it or not, I can get the number in international format anyway. So this part is fine.

However, from my research I've found that when talking about cell phone numbers, common practice seems to be to enter in international format regardless of if it is a local or international bound call. That is what I quoted in the first post in this thread. In either case, whether the user wants to enter their cell phone number in national or international format, the plugin works fine.

The reason why I brought up setNumber is because I was trying to say this: When I call `getNumber` I am able to request the output in a particular format. But when I use `setNumber`, it always displays the resulting output as national format. So even if I had saved the number in international format, the user is shown the national format instead of the format in which the number is being stored.

So basically the user may enter the number in either national or international format, but it always comes out in national format with no way of configuring it to do anything else.

It's really not that big of a deal, and it's not a blocker in any way. It's more of a cosmetic issue that I was trying to figure out because I feel like if I could tell the setNumber method which format the number I'm passing in is in, then it would display in a consistent manner. 

*TL;DR* - Right now if the user enters it in national format, and I save it in international format, when they view their user profile it shows up in national format again. There is no way for me to show the user that the number I am actually using is in international format.

**Comment by jackocnr** [comment]
> common practice seems to be to enter in international format
I originally assumed that (I'm from the UK), but since moving to America, I have found that most people here are not used to thinking about international dial codes.

As for your comments on `setNumber` I guess it might be nice to accept a 2nd argument for the format - I agree that it should be possible to present the number back to the user in the same format they entered it.

One thing to note here though is that currently the idea is you just store 1 thing: the full international number, which is what you need to send SMSs etc, but if you wanted to display the number to the user in the same format as they originally entered it, then you would also have to store a 2nd piece of information: the format.

**Comment by rderharo** [comment]
I'm less concerned about showing the number to the user as they *inputted* it, but more about showing the number to the user as I am *storing* it (because it is in this format that any SMS messages will be sent).

In my application, when I call `getNumber` I always pass the format as international, so I always store numbers in international format. This is why I don't store the format, because I like the fact that the user is able to type their number in whatever format is familiar to them and the plugin can still give it to me in the format I want.

So regardless of if they enter the number as:
917 555 1111
or 
0 917 555 1111
The way I store it is always going to be as:
+63 917 555 1111

Subsequently, when the user views their profile I would want them to see the number as it is stored:
917 555 1111 with the country dropdown selected

Like I said, it's just a minor visual thing.

**Comment by jackocnr** [comment]
Cool yeah I'd be open to a pull request for adding a second argument to `setNumber` which accepts one of those `intlTelInputUtils.numberFormat` enums. If no value is passed for that argument, it should default to NATIONAL if in nationalMode, else INTERNATIONAL.

**Comment by rderharo** [comment]
Great, thanks! Sorry for the delayed response. I hope to get something going soon.

**Comment by jackocnr** [comment]
I had some free time so I added this in v5.7. Note that it will only work with valid numbers. Let me know if you have any problems.

**Comment by rderharo** [comment]
Oh, thank you! I will make a note to upgrade to 5.7 this week to test it out.
